### ISSUE TITLE
Fatal python error / segfault for transformer training and PyTorch v1.13.0

### ISSUE THREAD

**Issue by adrianeboyd** [explosion/spaCy#11742]
We are aware of a bug related to PyTorch v1.13.0 that can lead to crashes when training a transformer model. We've seen crashes locally and in our CI for both linux and windows and for both CPU and GPU versions of `torch`.

Our recommended workaround is to downgrade to torch v1.12.1 as described here: https://pytorch.org/get-started/previous-versions/#v1121

We have reported this upstream and it currently looks like a future release of `numpy` should address this issue. We'll update this issue when a fix is released.

--------

In linux the crash can look like this:

```
Fatal Python error: PyEval_SaveThread: NULL tstate
Python runtime state: finalizing (tstate=0x181fea0)

Current thread 0x00007fc1f3d44740 (most recent call first):
<no Python frame>
Aborted (core dumped)
```

or

```
Fatal Python error: PyEval_SaveThread: the function must be called with the GIL held, but the GIL is released (the current Python thread state is NULL)
Python runtime state: finalizing (tstate=0x0000564d3aea8338)

Current thread 0x00007fd9191224c0 (most recent call first):
  Garbage-collecting
  <no Python frame>
Aborted (core dumped)
```

In windows it might look more like a generic segfault message:

```
##[error]Cmd.exe exited with code '-1073740791'.
```

**Comment by adrianeboyd** [comment]
Another possible workaround as of yesterday is to use a pre-release version of numpy v1.24:

```bash
pip install -U --pre numpy
```

**Comment by adrianeboyd** [comment]
This should be resolved by the release of numpy v1.24.0.

In addition, we have submitted a PR that fixes this behavior in pytorch for buggy versions of numpy (v1.22.x-v1.23.x), but we're not sure which version of pytorch will include the fix yet: https://github.com/pytorch/pytorch/pull/89759
